File: 04-labs/2026-09-07-lab3-crash-workshop/solution/04_type.py
# 04 -- SOLUTION: decimal="," was missing.
# Without it primaryValue is a text column -- and sum() on text raises NO error
# in Python: it concatenates, and returns a 6,349-character string. R refuses.
# The more permissive language is not the safer one.
import pandas as pd
trade = pd.read_csv("data/raw/comtrade_fr_roundwood_dirty.csv",
                    sep=";", decimal=",", encoding="latin-1")
print("Total value:", trade.primaryValue.sum(), "USD")

File: 04-labs/2026-09-07-lab3-crash-workshop/solution/04_type.R
# 04 -- SOLUTION: sep, dec AND fileEncoding. All three must be declared.
#   without dec=","       -> primaryValue read as text, sum() refuses
#   without fileEncoding  -> accents unreadable, and no error raised at all
trade <- read.csv("data/raw/comtrade_fr_roundwood_dirty.csv",
                  sep = ";", dec = ",", fileEncoding = "latin1")
cat("Total value:", sum(trade$primaryValue, na.rm = TRUE), "USD\n")
